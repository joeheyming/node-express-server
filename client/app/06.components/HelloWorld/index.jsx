import React from 'react';
import PropTypes from 'prop-types';

const HelloWorld = ({ message }) => {
  return (
    <div className="helloworld">
      <h1 className="helloworld__message">{message}</h1>
    </div>
  );
};

HelloWorld.propTypes = {
  message: PropTypes.string
};

HelloWorld.defaultProps = {
  message: 'Hello World!'
};

export default HelloWorld;
