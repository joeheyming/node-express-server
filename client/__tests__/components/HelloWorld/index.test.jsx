import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import HelloWorld from '../../../app/06.components/HelloWorld';

describe('HelloWorld', () => {
  describe('init component', () => {
    test('shoul render the component', () => {
      const wrapper = shallow(<HelloWorld />);
      expect(wrapper).toHaveLength(1);
    });

    test('should match the snapshot', () => {
      const wrapper = shallow(<HelloWorld />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('message props', () => {
    test('should render a message from props', () => {
      const wrapper = mount(<HelloWorld />);
      expect(wrapper.prop('message')).toBeDefined();
    });

    test('should render default message', () => {
      const message = undefined;
      const wrapper = mount(<HelloWorld message={message} />);
      expect(wrapper.prop('message')).toEqual('Hello World!');
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('should render a specific message', () => {
      const message = 'Hello Boilerplate!';
      const wrapper = mount(<HelloWorld message={message} />);
      expect(wrapper.prop('message')).toEqual(message);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
