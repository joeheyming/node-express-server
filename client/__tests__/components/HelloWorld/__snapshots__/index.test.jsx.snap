// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld init component should match the snapshot 1`] = `
<div
  className="helloworld"
>
  <h1
    className="helloworld__message"
  >
    Hello World!
  </h1>
</div>
`;

exports[`HelloWorld message props should render a specific message 1`] = `
<HelloWorld
  message="Hello Boilerplate!"
>
  <div
    className="helloworld"
  >
    <h1
      className="helloworld__message"
    >
      Hello Boilerplate!
    </h1>
  </div>
</HelloWorld>
`;

exports[`HelloWorld message props should render default message 1`] = `
<HelloWorld
  message="Hello World!"
>
  <div
    className="helloworld"
  >
    <h1
      className="helloworld__message"
    >
      Hello World!
    </h1>
  </div>
</HelloWorld>
`;
